### PR TITLE
documentation: adding change log for smd model parallel

### PR DESCRIPTION
**Pull request by TEChopra1000**
*Description of changes:*

* Adding section to SageMaker distributed model parallel to link to release notes 
* Adding folder with SMD model parallel release notes. 

*Testing done:*
* `tox -e black-check,flake8,pylint,docstyle,sphinx,doc8 --parallel all`

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by anirudh2290 — DISMISSED** [review]
LGTM!

File: doc/api/training/smd_model_parallel.rst [hunk 1]
```diff
@@ -20,6 +20,21 @@ Use the following sections to learn more about the model parallelism and the lib
    <https://integ-docs-aws.amazon.com/sagemaker/latest/dg/model-parallel-use-api.html#model-parallel-customize-container>`__
    for more information.
 
+How to Use this Guide
+=====================
+
+The library contains a Common API that is shared across frameworks, as well as APIs
+that are specific to supported frameworks, TensorFlow and PyTroch. To use the library, reference the
```
Comment (IvyBazan) [inline doc/api/training/smd_model_parallel.rst:27]: PyTorch

File: doc/api/training/smd_model_parallel.rst [hunk 2]
```diff
@@ -20,6 +20,21 @@ Use the following sections to learn more about the model parallelism and the lib
...
+**Common API** documentation alongside framework specific API documentation.
```
Comment (IvyBazan) [inline doc/api/training/smd_model_parallel.rst:28]: alongside the framework

File: doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    loss = model(custom_obj)
+    model.backward(loss)
+    return loss
+```
+
+### PyTorch
+
+#### Add support for smp.DistributedModel.cpu()
+
+`smp.DistributedModel.cpu()` allgathers parameters and buffers across all `mp_ranks` and moves them to the CPU.
```
Comment (ngluna) [inline doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.md:49]: "allgathers" ?

File: doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+## New Features
+
+The following sections describe new feature releases that are common across frameworks and that are framework specific.
+
+### Common across frameworks
+
+#### Custom slicing support (smp_slice method) for objects passed to smp.step decorated functions
+
+To pass an object to `smp.step` that contains tensors that needs to be split across
```
Comment (ngluna) [inline doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.md:17]: Bit of a run-on .

**Review by ngluna — DISMISSED** [review]

File: doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.md (hunk 3 above)
Comment (IvyBazan) [inline doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.md:49]: "allgathers" not sure what this should be

File: doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+### PyTorch
+
+#### Add support for smp.DistributedModel.cpu()
+
+`smp.DistributedModel.cpu()` allgathers parameters and buffers across all `mp_ranks` and moves them to the CPU.
+
+#### Add `trace_memory_usage` option to `smp.DistributedModel` to measure memory usage per module
+
+Adds `trace_memory_usage` option to `smp.DistributedModel`. This attempts to measure memory usage per module during
+tracing. If this is disabled, memory usage will be estimated through the sizes of tensors returned from the module.
```
Comment (IvyBazan) [inline doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.md:54]: memory usage is estimated

File: doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+This option is disabled by default.
+
+## Bug Fixes
+
+### PyTorch
+
+- `torch.nn.Sequential`: Fix a bug with `torch.nn.Sequential` which causes a failure with the error message : `shouldnt go less than 0, there is a bug` when the inputs to the first module don't require grads.
```
Comment (IvyBazan) [inline doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.md:61]: Use of ":" and "-" is inconsistent

**Review by IvyBazan — DISMISSED** [review]
Fix minor issues, but LGTM other than that

File: doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.md (hunk 3 above)
Comment (anirudh2290) [reply doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.md:49]: You can link to this documentation for allgather : https://sagemaker.readthedocs.io/en/stable/api/training/smd_model_parallel_common_api.html#smp.allgather

**Review by ngluna — APPROVED** [review]

File: doc/api/training/smd_model_parallel.rst (hunk 1 above)
Comment (easyj2j) [inline doc/api/training/smd_model_parallel.rst]: TensorFlow and **PyTroch**

**Pull request merged**